### PR TITLE
Migration from httplib2 to requests

### PR DESCRIPTION
**Pull request by sfultariya-crest**
Changes:

- feat: migration from httplib2 to requests library
- BREAKING CHANGE: the 'build_http_connection function of alert_actions_base.py has been deprecated. Users can use 'send_http_request' function of the same class or use 'requests.request' function.
- Changes related to 'splunkd_request' and 'code_to_msg' function of splunktalib.rest (of [this](https://github.com/splunk/addonfactory-ta-library-python/pull/48) PR) has been included in this PR.

**Changes needs to be done before merging this PR:**

- Need to update the version of splunktalib (comprises the changes of [this](https://github.com/splunk/addonfactory-ta-library-python/pull/48) PR) in the pyproject.toml file.

### CODE REVIEW

File: splunktaucclib/rest_handler/base.py [hunk 1]
```diff
@@ -68,15 +67,15 @@ def user_caps(mgmt_uri, session_key):
...
 
-    resp, cont = splunkd_request(
+    resp = splunkd_request(
         url, session_key, method="GET", data={"output_mode": "json"}, retry=3
     )
     if resp is None:
         RH_Err.ctl(500, logging.ERROR, "Fail to get capabilities of sessioned user")
-    elif resp.status not in (200, "200"):
-        RH_Err.ctl(resp.status, logging.ERROR, cont)
+    elif resp.status_code not in (200, "200"):
```
Comment (arys-splunk) [inline splunktaucclib/rest_handler/base.py:75]:
```suggestion
    elif resp.status_code != 200:
```

File: splunktaucclib/rest_handler/util.py [hunk 2]
```diff
@@ -95,3 +98,44 @@ def remove_http_proxy_env_vars():
             del os.environ[k]
         elif k.upper() in os.environ:
             del os.environ[k.upper()]
+
+
+def get_proxy_uri(proxy) -> Optional[str]:
```
Comment (arys-splunk) [inline splunktaucclib/rest_handler/util.py:103]: Let's at least say that `proxy` is `Dict[str, Any]`

**Review by arys-splunk — CHANGES_REQUESTED** [review]

File: splunktaucclib/rest_handler/base.py (hunk 1 above)
Comment (sfultariya-crest) [reply splunktaucclib/rest_handler/base.py]: Updated the code

File: splunktaucclib/rest_handler/util.py (hunk 2 above)
Comment (sfultariya-crest) [reply splunktaucclib/rest_handler/util.py]: Updated the code

File: splunktaucclib/config.py [hunk 3]
```diff
@@ -365,7 +356,8 @@ def load_value(self, endpoint_id, item_name, fname, fval):
 
     def try_fix_corrupted_json(self, corrupted_json, value_err):
         """
-        When L3 (one of our customers) was testing, they encountered a bug that 'access_token_encrypted' or 'refresh_token' got corrupted when it was saved in the conf file
+        When L3 (one of our customers) was testing, they encountered a bug that 'access_token_encrypted'
```
Comment (arys-splunk) [inline splunktaucclib/config.py:359]: Can you please remove that mention of our customer?

**Review by arys-splunk — COMMENTED** [review]
Looks good, I would like to have another pass from myself, then I'll approve.

File: splunktaucclib/config.py (hunk 3 above)
Comment (sfultariya-crest) [reply splunktaucclib/config.py]: Updated the docstring

**Comment by arys-splunk** [comment]
@sfultariya-crest Can you please update the branch? I've merged 2 PRs into main.

**Comment by sfultariya-crest** [comment]
> @sfultariya-crest Can you please update the branch? I've merged 2 PRs into main.

Updated the branch

**Comment by sfultariya-crest** [comment]
Closing this PR due to signed commit issues and changes have been added as part of PR #115

**Pull request closed without merging**